db/view/view.cc

```diff
@@ -2471,6 +2472,76 @@ future<> view_builder::do_build_step() {
     });
 }
 
+future<> view_builder::migrate_to_v2(db::system_keyspace& sys_ks, cql3::query_processor& qp, service::raft_group0_client& group0_client, abort_source& as, service::group0_guard guard) {
+    // TODO MICHAEL
+    // currently we don't handle concurrent writes to the table while we migrate it,
+    // and we may lose writes.
+    // maybe we can wait until there are no view builds in progress and assume no new
+    // view builds start while we migrate.
```

Comment:
I think currently it's possible we lose writes to the table during the migration. I'm not sure how it would affect things and whether we should consider a more robust migration scheme.
By my understanding the table is used to decide whether we should generate view updates for streaming tables that have view building in progress.
So possibly this could cause us to not generate view updates in this scenario, or (less critical) generate unnecessary updates.

Reply:
> I think currently it's possible we lose writes to the table during the migration. I'm not sure how it would affect things and whether we should consider a more robust migration scheme.
> By my understanding the table is used to decide whether we should generate view updates for streaming tables that have view building in progress.
> So possibly this could cause us to not generate view updates in this scenario, or (less critical) generate unnecessary updates.

To not lose newer writes, we write both to the source as well as the destination during migration. Yes this status table is used to - resume view update work on destination after topology changes. We only need to make sure the destination registers the sstables in staging state to the view update generator. Naturally once the read selector for a token points to the new destination replica, the view updates will no longer happen from the source. There's a GitHub issue opened for that. I will later reference it here

Reply:
https://github.com/scylladb/scylladb/issues/19149

Reply:
I added a new commit with additional intermediate migration phase where we write to both tables and wait for old operations in order to not lose writes.